src/main/java/invtweaks/container/DirectContainerManager.java

```diff
@@ -82,8 +82,8 @@ public boolean move(ContainerSection srcSection, int srcIndex, ContainerSection
 
         // Use intermediate slot if we have to swap tools, maps, etc.
         assert srcStack != null;
-        if(destStack != null && srcStack.getItem() == destStack.getItem() && (srcStack.getMaxStackSize() == 1 ||
-                srcStack.hasTagCompound() || destStack.hasTagCompound())) {
+        if(destStack != null && srcStack.isItemEqual(destStack) && srcStack.getMaxStackSize() == 1 &&
+                ((srcStack.hasTagCompound() || destStack.hasTagCompound()) && (srcStack.getTagCompound().equals(destStack.getTagCompound())))) {
```

Comment:
Like I said, I think the conditional here is a bit awkward.

I need to take a bigger look at the whole code and see if it can just be simplified, it seems to be a little too specific at the moment.